### PR TITLE
Factor out creation of DatastoreService (alternate gNMI service)

### PR DESCRIPTION
**Pull request by wenovus**

### CODE REVIEW

File: gnmi/gnmit/gnmit.go [hunk 1]
```diff
@@ -181,6 +181,27 @@ func NewServer(ctx context.Context, hostname string, sendMeta bool, tasks []Task
 	return c, gnmiserver, nil
 }
 
+func StartDatastoreServer(gnmiServer *GNMIServer) (func(), error) {
```
Comment (DanG100) [inline gnmi/gnmit/gnmit.go:184]: add comment (especially clarifying return values

File: gnmi/gnmit/gnmit.go [hunk 2]
```diff
@@ -181,6 +181,27 @@ func NewServer(ctx context.Context, hostname string, sendMeta bool, tasks []Task
...
+	if err := os.RemoveAll(DatastoreAddress); err != nil {
+		log.Fatal(err)
```
Comment (DanG100) [inline gnmi/gnmit/gnmit.go:186]: return err instead

**Review by DanG100 — APPROVED** [review]

File: gnmi/gnmit/gnmit.go (hunk 1 above)
Comment (wenovus) [reply gnmi/gnmit/gnmit.go:184]: oops I put the comment in the subsequent PR.

File: gnmi/gnmit/gnmit.go (hunk 2 above)
Comment (wenovus) [reply gnmi/gnmit/gnmit.go:186]: Done.

**Pull request merged**